clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala

```diff
@@ -147,22 +147,28 @@ trait PersistenceDAO extends LazyLogging {
     lastToIgnore: Option[Path]
   )(implicit ec: ExecutionContext, materializer: Materializer): Source[D, NotUsed] = {
 
-    val dayFilter = lastToIgnore.fold((_: Path) => true) {
+    val dayFilter = lastToIgnore.fold((_: Path) => true) { last =>
+      val dayDirectoryOfLastUpsert = last.getParent.toString
       /*
-       * We filter inclusively on date because the most recent
-       * document in Elasticsearch might not have been the last
-       * document upsert-ed during the day it was added.
+       * Given the path in storage to the last upsert known to Elasticsearch, we want to keep
+       * only the day-directory in which we find the upsert, and all following directories.
        */
-      last => (dir: Path) =>
-        last.getParent.toString >= dir.toString
+      _.toString >= dayDirectoryOfLastUpsert
     }
 
-    val docFilter = lastToIgnore.fold((_: Path) => false) { last => (path: Path) =>
-      last.toString >= path.toString
+    val docIsNotMoreRecentThanLastUpsert = lastToIgnore.fold((_: Path) => false) { last =>
+      val pathOfLastUpsert = last.toString
+      _.toString <= pathOfLastUpsert
     }
 
+    /*
+     * `dropWhile` instead of `filterNot` because we know `getPathsOrderedBy` is going
+     * to return paths in order by time, so once we see a document which is more recent
+     * than the latest known upsert, all following documents must also be more recent
+     * than that upsert.
+     */
```

Comment:
... because getPathsOrderedBy returns the paths sorted from earliest to latest, so this doesn't have to test paths after it finds one more recent than the latest upsert.